include/yaml-cpp/node/convert.h

```diff
@@ -161,83 +167,272 @@ struct convert<bool> {
   YAML_CPP_API static bool decode(const Node& node, bool& rhs);
 };
 
-// std::map
-template <typename K, typename V>
-struct convert<std::map<K, V>> {
-  static Node encode(const std::map<K, V>& rhs) {
+template <class T> class node_push_backer
```

Comment:
I think these helper classes obfuscate what's going on here. Can you just inline the loops instead?